### PR TITLE
Child/parent code lenses and label parent in dependsOn completions

### LINKED ISSUE #943: Show code lenses for resource parent and children
- [x] Can you have a decoupled parent/child not at the top level?
- [x] Put it inside the curly braces?
- [x] don't select entire resource
- [x] option to disable
- [x] click
  - [x] child of: navigate to parent
  - [x] children: show refs

### PR DESCRIPTION
**Pull request by StephenWeatherford**
Depends on https://github.com/microsoft/vscode-azurearmtools/pull/959
Fixes #943 

image
image
image

image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -141,12 +141,17 @@
...
                 },
                 "azureResourceManagerTools.codelens.parameters": {
                     "type": "boolean",
-                    "description": "Enables code lens for parameter definitions and values",
+                    "description": "Enables Code Lens for parameter definitions and values",
+                    "default": true
+                },
+                "azureResourceManagerTools.codelens.resourceChildren": {
+                    "type": "boolean",
+                    "description": "Enables Code Lens that displays the parent and children each resource",
```
Comment (bhsubra) [inline package.json:154]: nit: Enables Code Lens that displays the parent and children for each resource

File: src/documents/templates/ParentAndChildCodeLenses.ts [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+                            }
+                        }
+                    ]
+            };
+        } else {
+            this.command = {
+                title: title,
+                command: ''
+            };
+
```
Comment (bhsubra) [inline src/documents/templates/ParentAndChildCodeLenses.ts:66]: nit: remove empty line

File: src/documents/templates/ParentAndChildCodeLenses.ts [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+                            telemetryProperties: {
+                                kind,
+                                targetCount: String(targets.length)
+                            }
+                        }
+                    ]
+            };
+        } else {
+            this.command = {
+                title: title,
```
Comment (bhsubra) [inline src/documents/templates/ParentAndChildCodeLenses.ts:63]: Just curious. How is this rendered and used?

Comment (StephenWeatherford) [reply src/documents/templates/ParentAndChildCodeLenses.ts:63]: See the images in the description, e.g.:
image

The command title is the "No parent" or "2 children: xxx" text.  If you click on a command with children or parent, it peeks them in the editor.
image

File: src/documents/templates/ParentAndChildCodeLenses.ts (hunk 2 above)
Comment (StephenWeatherford) [reply src/documents/templates/ParentAndChildCodeLenses.ts:66]: I tend to put blank lines after if/then's or anywhere else it makes sense to break the flow of logic.

**Comment by StephenWeatherford** [comment]
@bhsubra Could you approve this one today?  thanks.

**Pull request merged**